### ISSUE TITLE
[a11y Design Review]: Veteran Support Team: Feedback Mechanism - Rate Your Experience

### ISSUE THREAD

**Issue by sara-amanda** [department-of-veterans-affairs/va.gov-team#67857]
## Auditor(s)

@sara-amanda 
@SarahKay8 

## Important Notes
- **Focus:** There may be potential _**Medallia constraints**_
- **Midpoint Review:** Async on Friday Oct. 20, 2023, from 2-2:30 p.m. (source Collab Cycle Calendar)
- **Review Figma Files:** Provided by Naomi on 10/17/2023, (added to Prototype section of this ticket)

<details><summary>Toggle Design Intent Accessibility Feedback</summary>

## Governance Feedback

Practice areas will document their feedback on the VFS-provided artifacts following the [Must, Should, and Consider Framework](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/must-should-and-consider-framework-for-feedback). Platform reviewers may also provide additional notes that don’t comment on the artifacts themselves but are important for implementation (eg. engineering/coding notes).

### Must
Biggest concern here is that when content on the page changes (second-step questions revealed for the "bad" flow, and the thank you message for both flows), those changes are somehow communicated to screen reader users. We don't usually get into this much implementation detail at a Design Intent, but since you're working with a third-party on this I want to give you some notes for testing and/or sharing with Medallia sooner rather than later.

From what's been shared so far, I think what I would expect to happen would be something like:

#### User selects "Bad."
"Report a problem on this page" fieldset is revealed.

- User presses [tab] to move to the next focusable element, which is the first checkbox. If "Report a problem on this page" is the legend for the fieldset, a screen reader announces both the legend and the label for the first checkbox.
- User selects the applicable checkboxes and enters anything they want to share in "Other"
- User presses [tab] to move to the Submit Feedback button.
- User presses [enter] to submit.
- Keyboard focus moves to the "Thank you, your feedback has been submitted" text (I'm guessing a p with tabindex="-1").

### Key questions for Medallia would be:

- Are the checkboxes grouped together in a fieldset with the question in a legend?
- Are checkbox labels associated with the checkboxes?
- Where does focus go after pressing Submit Feedback?

### Should
I think I have a preference for the "Other" field to be its own standalone question, so that it makes sense in isolation separate from the checkboxes. Just a more descriptive label, like "Is something else wrong with this page?" or something similar. I don't think I would expect the "Other" field to be grouped inside the same fieldset as the checkboxes, which means screen reader users could potentially tab into it without the context of "Report a problem on this page" to understand what it's for.
</details>

## Design Review
<details><summary>Toggle to View Design Review Testing Areas</summary>

Based on the development of the prototype, some of the following may or may not be applicable. 
**This list has been moved to tasks in this ticket.**
</details>

## Prototype Links
<details><summary>Toggle to View Prototype Links</summary>

### Test with Figma Prototypes
- [Sketch File (from original intake ticket)](https://www.sketch.com/s/a513ba38-1d26-41d0-9009-92ea729446e2)
- **Requested Review of the Figma Prototypes** `c/o Naomi 10/17/2023 `
    - Figma: [Desktop Prototype](https://www.figma.com/proto/s9aAXodKPxFE8ffYO8vg0T/Rate-tool?page-id=0%3A1&type=design&node-id=121-1632&viewport=-131%2C1063%2C0.23&t=BQJ9kRzugzISzh2Q-1&scaling=min-zoom&starting-point-node-id=121%3A1632&show-proto-sidebar=1)
    - Figma: [Mobile Prototype](https://www.figma.com/proto/s9aAXodKPxFE8ffYO8vg0T/Rate-tool?page-id=83%3A1743&type=design&node-id=182-5100&viewport=1097%2C286%2C0.03&t=nqbxGBXKNBTo5igg-1&scaling=scale-down&starting-point-node-id=91%3A2604)
</details>

## Findings & Recommendations from CAIA A11ys

- [x] Findings Identified:
    - [x] Major (defects 0-2): 2
    - [x] Minor (defect 4): 1
- [ ] No Findings Identified

<details><summary>Toggle to Reveal Defect Findings from CAIA A11ys</summary>

## Major (defects 0-2)
- **Heading Order** 
    - #67861
- **Duplicate Checkbox** 
    - #67876

## Minor (defect 4)

- **Language (ease of reading)**
    - #67883

</details>

## Testing Tasks for CAIA A11ys 

**The following are the areas that we review for the CAIA A11y Design Review.** 
- [x] As items are reviewed boxes will be checked.  
- [x] If there is a finding, an issue will be created and linked in that task list. 

**_Based on the development of the prototype, some of the following may or may not be applicable._** 

```[tasklist]
### a11y Design Review: Color and Contrast Checks Mobile & Desktop
- [x] [Color and contrast checks](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#use-of-color)
- [x] WCAG 2.1 AA
- [x] WCAG 2.1 AAA
- [x] [Who Can Use](https://www.whocanuse.com/?b=ecece5&c=000000&f=20&s=)
- [x] High Contrast Mode
```
```[tasklist]
### a11y Design Review: Keyboard Navigation
- [x] [Keyboard navigation](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review#keyboard-nav) Mobile & Desktop
- [x] Annotations for Tab order, reflow and reading order  Mobile & Desktop **(if needed)**
```
```[tasklist]
### a11y Design Review: Accessible Name Annotations & Language
- [x] [Alt attributes for images](https://webaim.org/techniques/alttext/) Mobile & Desktop
- [x] Text alternative for complex images such as charts graphs maps etc Mobile & Desktop
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/67883
```
```[tasklist]
### a11y Design Review: HTML Tags & Size of Clickable Areas
- [x] **HTML Tags:** [Page Title](https://accessibility.18f.gov/page-titles/) Mobile & Desktop
- [x] **HTML Tags:** [Heading order](https://accessibility.18f.gov/headings/) Mobile & Desktop
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/67861
- [x] **HTML Tags:** Annotations (if needed)
- [x] **HTML Tags:** Semantic Consideration for paragraphs ordered lists numbering Mobile & Desktop
- [x] **HTML Tags:** [Tables (heading rows)](https://design.va.gov/components/table) Mobile & Desktop
- [x] Size of clickable areas Mobile & Desktop
- [ ] https://github.com/department-of-veterans-affairs/va.gov-team/issues/67876
```
```[tasklist]
### a11y Design Review: As Time Allows
- [x] [Components](https://design.va.gov/components/) follow VA design system guidance.
- [x] [Patterns](https://design.va.gov/patterns/) follow VA design system guidance.
```

**Comment by SarahKay8** [comment]
![image](https://github.com/department-of-veterans-affairs/va.gov-team/assets/102324990/77f2fd30-d35b-4187-b024-aa05e05172b7)

"Text is hard to understand" `checkbox` is listed here twice

**Comment by sara-amanda** [comment]
## Closing Out Ticket: Design Review
_We can open another ticket, if additional support is requested, post-midpoint (10/20/2023)._ 
- [Posted findings in ticket.](https://github.com/department-of-veterans-affairs/va.gov-team/issues/67857#:~:text=No%20Findings%20Identified-,Toggle%20to%20Reveal%20Defect%20Findings%20from%20CAIA%20A11ys,-Testing%20Tasks%20for) 
- [Tagged findings in comments.](https://github.com/department-of-veterans-affairs/va.gov-team/issues/67857#issuecomment-1769361642) 
    - Closing 2 of 3 findings: 
        - #67876
        - #67861
    - One finding may need potential CAIA Content support. Added to intake content task list @strelichl 
        - #67883
- [Shared via Slack](https://dsva.slack.com/archives/C061RPL9LTT/p1697665239405259). 

cc: @coforma-terry
